pkg/controller/scyllacluster/sync_certs.go

```diff
@@ -0,0 +1,205 @@
+) (*scyllav1.ScyllaClusterStatus, error) {
+	// Be careful to always apply signers first to be sure they have been persisted
+	// before applying any child certificate signed by it.
+
+	var errors []error
+
+	cm := okubecrypto.NewCertificateManager(
+		scc.kubeClient.CoreV1(),
+		scc.secretLister,
+		scc.kubeClient.CoreV1(),
+		scc.configMapLister,
+		scc.eventRecorder,
+	)
+
+	clusterLabels := naming.ClusterLabels(sc)
+
+	// Manage client certificates.
+	errors = append(errors, cm.ManageCertificates(
+		ctx,
+		time.Now,
+		&sc.ObjectMeta,
+		scyllaClusterControllerGVK,
+		&okubecrypto.CAConfig{
+			MetaConfig: okubecrypto.MetaConfig{
+				Name:   naming.GetScyllaClusterLocalClientCAName(sc.Name),
+				Labels: clusterLabels,
+			},
+			Validity: 10 * 365 * 24 * time.Hour,
+			Refresh:  8 * 365 * 24 * time.Hour,
+		},
+		&okubecrypto.CABundleConfig{
+			MetaConfig: okubecrypto.MetaConfig{
+				Name:   naming.GetScyllaClusterLocalClientCAName(sc.Name),
+				Labels: clusterLabels,
+			},
+		},
+		[]*okubecrypto.CertificateConfig{
+			{
+				MetaConfig: okubecrypto.MetaConfig{
+					Name:   naming.GetScyllaClusterLocalUserAdminCertName(sc.Name),
+					Labels: clusterLabels,
+				},
+				Validity: 30 * 24 * time.Hour,
+				Refresh:  20 * 24 * time.Hour,
+				CertCreator: (&ocrypto.ClientCertCreatorConfig{
+					Subject: pkix.Name{
+						CommonName: "",
+					},
+					DNSNames: []string{"admin"},
+				}).ToCreator(),
+			},
+		},
+		secrets,
+		configMaps,
+	))
+
+	// Manage serving certificates.
+	// We are currently using StatefulSet that allows only a uniform config so we have to create a multi-SAN certificate.
+	// TODO: Create dedicated certs when we get rid of StatefulSets.
+	// TODO: we need to somehow check the domain given so we don't start signing stuff for e.g. google.com.
+	//       (maybe cluster domains CR, or forcing a suffix). On the other hand the CA should be trusted
+	//       only for the connection, not by the system.
+
+	// For the clients not using the proxy we'll sign for IPs that are published by ScyllaDB.
+	var ipAddresses []net.IP
+	for _, svc := range serviceMap {
+		if svc.Labels[naming.ScyllaServiceTypeLabel] != string(naming.ScyllaServiceTypeMember) {
+			continue
+		}
+
+		ips := svc.Spec.ClusterIPs
+		if len(ips) == 0 {
+			klog.V(2).InfoS("No IP available for member", "Service", klog.KObj(svc))
+			continue
+		}
+		parsedIPs, err := helpers.ParseClusterIPs(ips)
+		if err != nil {
+			return status, err
+		}
+
+		ipAddresses = append(ipAddresses, parsedIPs...)
+	}
+
+	var hostIDs []string
+	var progressingMessages []string
+	for _, svc := range serviceMap {
+		if svc.Labels[naming.ScyllaServiceTypeLabel] != string(naming.ScyllaServiceTypeMember) {
+			continue
+		}
+
+		hostID, hasHostID := svc.Annotations[naming.HostIDAnnotation]
+		if len(hostID) == 0 {
+			if hasHostID {
+				klog.Warningf("service %q has empty hostID", klog.KObj(svc))
+			}
+
+			// TODO: this needs to be in an outer loop or it complains twice.
```

Comment:
minor TODO that could be fixed now

Reply:
i think this is just a leftover from the times this was inside 2 for loops that got splitted later on

Reply:
removed